### PR TITLE
[Feature]Map read map from csv string

### PR DESCRIPTION
**Pull request by fzhedu**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

TODO:
column delimiter conflicts with map's delimiter, for example `{'c':0},{'a':1, 'b':2}` the second map is invalid. 
## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/formats/csv/converter.cpp [hunk 1]
```diff
@@ -75,6 +76,14 @@ static std::unique_ptr<Converter> create_converter(const TypeDescriptor& t, cons
         return std::make_unique<DecimalV3Converter<int128_t>>(t.precision, t.scale);
     case TYPE_JSON:
         return std::make_unique<JsonConverter>();
+    case TYPE_MAP: {
+        auto key_con = get_converter(t.children[0], true);
+        auto value_con = get_converter(t.children[1], true);
+        if (key_con == nullptr || value_con == nullptr) {
+            return nullptr;
```
[the remark this answers is not in the document]
Comment (fzhedu) [reply be/src/formats/csv/converter.cpp:83]: previous Line:67 also return null,so return null keep the same behavior.

File: be/src/formats/csv/map_converter.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+Status MapConverter::write_quoted_string(OutputStream* os, const Column& column, size_t row_num,
+                                         const Options& options) const {
+    return write_string(os, column, row_num, options);
+}
+
+bool MapConverter::validate(const Slice& s) const {
+    if (s.size < 2) {
+        return false;
+    }
+    if (s[0] != '{' || s[s.size - 1] != '}') {
```
[the remark this answers is not in the document]
Comment (fzhedu) [reply be/src/formats/csv/map_converter.cpp:54]: strings have already been trimed, like in arrayConverter.

File: be/src/formats/csv/map_converter.cpp [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        return true;
+    }
+
+    bool in_quote = false;
+    int map_nest_level = 0;
+    int last_index = 0;
+    size_t i = 0;
+    for (; i < s.size; i++) {
+        char c = s[i];
+        if (c == '"') {
```
[the remark this answers is not in the document]
Comment (fzhedu) [reply be/src/formats/csv/map_converter.cpp:74]: `O(n)`

File: be/src/formats/csv/map_converter.cpp [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        return false;
+    }
+    size_t old_size = keys->size();
+    DCHECK_EQ(old_size, offsets->get_data().back());
+    DCHECK_EQ(old_size, values->size());
+
+    // get unique keys
+    std::vector<bool> unique_keys;
+    int unique_num = 0;
+    for (auto i = 0; i < key_fields.size(); ++i) {
```
[the remark this answers is not in the document]
Comment (fzhedu) [reply be/src/formats/csv/map_converter.cpp:125]: a map's size may not very large. Optimize it when it becomes a bottleneck in the future.

File: be/src/exprs/anyval_util.cpp [hunk 5]
```diff
@@ -142,6 +142,7 @@ FunctionContext::TypeDesc AnyValUtil::column_type_to_type_desc(const TypeDescrip
         break;
     case TYPE_FUNCTION:
         out.type = TYPE_FUNCTION;
+        break;
```
[the remark this answers is not in the document]
Comment (fzhedu) [reply be/src/exprs/anyval_util.cpp:145]: leak it at that time.

**Comment by fzhedu** [comment]
support collocation delimiter?

**Review by Seaven — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Pull request merged**